app/models/playlist.rb

```diff
@@ -18,15 +18,15 @@ class Playlist < ApplicationRecord
   enum access: { shared: 0, personal: 1, secret: 2 }
   enum playlist_type: { album: 0, artist: 1, track: 2 }
 
-  scope :with_item_ids, -> { left_joins(:items).select('playlists.*', 'array_agg(playlist_items.item_id ORDER BY playlist_items.order ASC) as item_ids').group(:id) }
+  scope :with_item_ids, -> { left_joins(:items).select('playlists.*', 'array_remove(array_agg(playlist_items.item_id ORDER BY playlist_items.order ASC), NULL) as item_ids').group(:id) }
 
   validates :name, presence: true
 
   before_save :normalize_item_order
 
   def item_ids
     # If with_item_ids was used, we use the output from the SQL query
-    return read_attribute(:item_ids) if has_attribute?(:item_ids)
+    return read_attribute(:item_ids) || [] if has_attribute?(:item_ids)
```

Comment:
Is there really no way to make the SQL query return an empty array?

Reply:
We could do this with the following, but lose some readability:
```sql
COALESCE(array_remove(array_agg(playlist_items.item_id ORDER BY playlist_items.order ASC), NULL), [])
```

Reply:
I think I prefer that solution. The SQL is already complicated enough that it needs some comments IMO, so adding some slight extra complexity there seems preferable to me to adding it here.